robotframework_reportportal/listener.py

```diff
@@ -188,6 +190,10 @@ def start_test(self, name, attributes, ts=None):
         :param attributes: Dictionary passed by the Robot Framework
         :param ts:         Timestamp(used by the ResultVisitor)
         """
+        if 'source' not in attributes:
+            # no 'source' parameter at this level for Robot versions < 4
+            attributes = attributes.copy()
```

Comment:
No need to make a copy here.

Reply:
These attributes were given by Robot Framework to us. We don't know how they were used or how they will be used. So to avoid unpredictable behavior modification I added this copying.

Reply:
Sorry did not get your answer.

Reply:
I mean listener interface is the last destination point for the attributes. But, if it looks safer to you, then leave it as is.

Reply:
I would leave it. I don't know how users might hack us, but better don't give a chance.